### PR TITLE
OSOE-553: Apply automatic package.json dependency installation and update NE package reference

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OSOE-553](https://lombiq.atlassian.net/browse/OSOE-553)

[OSOE-553]: https://lombiq.atlassian.net/browse/OSOE-553?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Lombiq.Privacy/package.json [hunk 1]
```diff
@@ -4,5 +4,13 @@
...
     "clean": "npm explore nodejs-extensions -- pnpm clean:styles",
     "watch": "npm explore nodejs-extensions -- pnpm watch:styles"
+  },
+  "devDependencies": {
+    "eslint": "8.25.0",
+    "eslint-config-airbnb-base": "15.0.0",
+    "eslint-plugin-import": "2.26.0",
+    "eslint-plugin-node": "11.1.0",
+    "eslint-plugin-only-warn": "1.0.3",
+    "eslint-plugin-promise": "6.1.0"
```
Comment (0liver) [inline Lombiq.Privacy/package.json]: These should and will contain the `^` prefix, once we switch from `pnpm install` to `pnpm add`. I'll the other submodules for now.

File: Lombiq.Privacy/Lombiq.Privacy.csproj [hunk 2]
```diff
@@ -51,7 +51,7 @@
 
   <ItemGroup Condition="'$(NuGetBuild)' == 'true'">
     <PackageReference Include="Lombiq.HelpfulLibraries.OrchardCore" Version="5.1.1" />
-    <PackageReference Include="Lombiq.NodeJs.Extensions" Version="1.0.0" />
+    <PackageReference Include="Lombiq.NodeJs.Extensions" Version="1.0.1-alpha.osoe-553.0" />
```
Comment (0liver) [inline Lombiq.Privacy/Lombiq.Privacy.csproj]: The version should be: `1.0.1-alpha.0.osoe-553`.

File: Lombiq.Privacy/Migrations/Recipes/PrivacyConsentCheckboxSettings.recipe.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
         "Owner": "",
         "Author": "admin",
         "LiquidPart": {
-          "Liquid": "<span>I've read and agree to the site's <a href='/privacy-policy' target='_blank'>privacy policy</a>.</span>"
+          "Liquid": "<span>I've read and agree to the site's <a href=\"{{ '~/privacy-policy' | href }}\" target=\"_blank\">privacy policy</a>.</span>"
```
Comment (0liver) [inline Lombiq.Privacy/Migrations/Recipes/PrivacyConsentCheckboxSettings.recipe.json:18]: What exactly does `{{ '~/privacy-policy' | href }}` do, and why did you change it in this PR?

**Review by 0liver — CHANGES_REQUESTED** [review]

File: Lombiq.Privacy/package.json (hunk 1 above)
Comment (sarahelsaig) [reply Lombiq.Privacy/package.json]: It appears you were looking at an outdated commit, I've already updated them with the carets [in this commit](https://github.com/Lombiq/Orchard-Privacy/pull/50/commits/f58c266181c7834ab3de4caf51290e89bfadf291). Same with the other submodules.

File: Lombiq.Privacy/Migrations/Recipes/PrivacyConsentCheckboxSettings.recipe.json (hunk 3 above)
Comment (sarahelsaig) [reply Lombiq.Privacy/Migrations/Recipes/PrivacyConsentCheckboxSettings.recipe.json:18]: The URL prefix should be resolved otherwise tenants will get the wrong address, so hard coded URL in the original is a bug, see [here](https://github.com/Lombiq/Lombiq-Tenants-Core/pull/3#discussion_r1102136992). I've changed it in this PR, because it had the same bug I've mentioned [here](https://github.com/Lombiq/Orchard-Vue.js/pull/75#issuecomment-1415664721) so we need this issue for the project.

File: Lombiq.Privacy/package.json (hunk 1 above)
Comment (0liver) [reply Lombiq.Privacy/package.json]: This was an old comment, that was published just now. So, yes, it refers to old code.

File: Lombiq.Privacy/Migrations/Recipes/PrivacyConsentCheckboxSettings.recipe.json (hunk 3 above)
Comment (0liver) [reply Lombiq.Privacy/Migrations/Recipes/PrivacyConsentCheckboxSettings.recipe.json:18]: I was wondering what the ` | href` part means in that template.

Comment (sarahelsaig) [reply Lombiq.Privacy/Migrations/Recipes/PrivacyConsentCheckboxSettings.recipe.json:18]: It's a built-in OC Liquid filter: https://docs.orchardcore.net/en/main/docs/reference/modules/Liquid/#href

**Review by 0liver — APPROVED** [review]
